core/storage/pager.rs

```diff
@@ -1015,11 +1115,14 @@ impl Pager {
                 page.set_dirty();
                 self.add_dirty(page.get().id);
 
-                let page_key = PageCacheKey::new(page.get().id);
+                let page_key = PageCacheKey(page.get().id);
                 let mut cache = self.page_cache.write();
                 match cache.insert(page_key, page.clone()) {
                     Ok(_) => (),
-                    Err(CacheError::Full) => return Err(LimboError::CacheFull),
+                    Err(CacheError::Full) => {
+                        drop(cache);
+                        self.stress(page)?;
+                    }
```

Comment:
Same thing with `read_page`, as this performs I/O we should make this re-entrant.